packages/app/src/cli/constants.ts

```diff
@@ -43,19 +43,6 @@ export const blocks = {
   },
 } as const
 
-/**
- * List of extensions for each category that are limited by quantity, only 1 of each is allowed per app
- */
-export const limitedExtensions: {
-  ui: UIExtensionTypes[]
-  theme: ThemeExtensionTypes[]
-  function: FunctionExtensionTypes[]
-} = {
-  ui: ['product_subscription', 'checkout_post_purchase', 'web_pixel_extension'],
-  theme: ['theme'],
-  function: [],
-}
-
 export const publicFunctionExtensions = {
```

Comment:
_ultimately_ the constants file should end up much lighter, maybe some nice type utilities to e.g. _extract_ the `ExtensionTypes` type from the specs

Reply:
We have a lot of `switch`es for extensions configuration that depend on the extension type, so I don't think it's possible to enable a new extension without CLI changes. [Example](https://github.com/Shopify/cli/blob/main/packages/app/src/cli/services/deploy.ts#L169-L202)

Since this switches receive a `UIExtensionType`, if we add new types that we don't know how to work with, it will force us to add dummy implementations. 

We shouldn't allow to scaffold cli-unsupported extensions, even if you have access to them, and I don't think that typescript can help with that, it should be a hardcoded list somewhere. (if i'm wrong and there is a typescript solution for that, happy to add it!)

Reply:
That's related to one of my open questions: when do we generate this file. My previous comment was under the assumption that it might be an automated process, but probably makes more sense to be a manual one? When would it be manually triggered? Per your comment, I guess that after the manual generation a manual validation step is required too (to not include unknown extensions)